fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsInteractor.kt

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations
+
+import mozilla.components.concept.engine.translate.TranslationOptions
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * Manages coordinating the functionality interactions for use in the views.
+ */
+class TranslationsInteractor(
+    private val translationUseCase: SessionUseCases.TranslateUseCase,
+) {
+    private val logger = Logger("TranslationsInteractor")
+
+    /**
+     * Translates the page on the given [tabId].
+     *
+     * @param tabId The ID of the tab to translate.
+     * @param fromLanguage The BCP-47 language code to translate from. Usually will be the detected
+     * page language.
+     * @param toLanguage The BCP-47 language code to translate to. Usually will be the user's
+     * preferred language.
+     * @param options Optional options to specify and additional criteria for the translation.
+     */
+    fun onTranslate(
+        tabId: String?,
+        fromLanguage: String,
+        toLanguage: String,
+        options: TranslationOptions?,
+    ) {
+        logger.info("Requesting a translation.")
+        translationUseCase.invoke(tabId, fromLanguage, toLanguage, options)
```

Comment:
I think the interactor should only call some methods from a controller, and the implementation of the methods should be done in the controller. So here, I think a controller is needed. Just like AddLoginInteractor and SavedLoginsStorageController.

Reply:
I think something like this it should look like : 
 interactor = TranslationsInteractor(
            TranslationsController(
                translationUseCase = requireComponents.useCases.sessionUseCases.translate,
                browserStore = requireComponents.core.store,
                tabId = args.sessionId,
            ),
        ) 
in Fragment.

Reply:
Thanks for taking a look at this! 

That makes sense to me for sure! I had started this as a controller in a draft and refactored it out after reading this [architecture document](https://github.com/mozilla-mobile/firefox-android/blob/93c3ebad015317fcefd831b66c43cc64d763d616/fenix/docs/architecture-overview.md#interactor) misinterpreting something. I'll move back to controller, that seemed right to me at the time too.

Rereading it now, I think I interpreted:
>Note that prior to the introduction of Controllers, Interactors handled the responsibilities of both objects. You may still find this pattern in some parts of the codebase, but it is being actively refactored out.

as "don't add new `Controllers`", but on rereading I see `actively refactored out` is talking about `Interactors handled the responsibilities of both objects`.

Reply:
Updated to a `controller` pattern, and thanks for the really clear login example!